app/src/main/java/org/mozilla/reference/browser/browser/CustomTabsIntegration.kt

```diff
@@ -74,6 +78,24 @@ class CustomTabsIntegration(
 
             SimpleBrowserMenuItem("Find in Page") {
                 FindInPageIntegration.launch?.invoke()
+            },
+
+            SimpleBrowserMenuItem("Open in Browser") {
+                // Release the session from this view so that it can immediately be rendered by a different view
+                engineView.release()
+
+                // Stip the CustomTabConfig to turn this Session into a regular tab and then select it
+                sessionManager.findSessionById(sessionId)?.let { session ->
+                    session.customTabConfig = null
+                    sessionManager.select(session)
+                }
+
+                // Close this activity since it is no longer displaying any session
+                activity?.finish()
+
+                // Now switch to the actual browser which should now display our new selected session
+                val intent = Intent(context, BrowserActivity::class.java)
+                context.startActivity(intent)
```

Comment:
This seems brittle. i.e. the `sessionManager` selecting the `session` then switching activity.

Reply:
That seems fine - what's your concern? :)

That may be something to hide behind a use case.